File: NetFramework/XHtmlEditor/VB/RichTextEditor/AppMenuTabs/OpenTab.vb

Imports RichTextEditor.RichTextEditor

Partial Public Class OpenTab
    Inherits AppMenuTabFileList

    Public Sub New()
        Button.ItemImage = Resources.Open_large
        Button.ItemText = Resources.Browse_txt
        Caption = GetElementText(Resources.OpenTab_txt)
    End Sub

    Protected Overrides Sub OnMenuItemClick(ByVal mi As AppMenuTabItem)
        Dim ribbon As C1TextEditorRibbon = CType(RibbonApplicationMenu.Ribbon, C1TextEditorRibbon)
        ribbon.OpenDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text))
        mi.Item.ItemDate = DateTime.Now
    End Sub

    Protected Overrides Sub OnMenuButtonClick()
        Dim ribbon As C1TextEditorRibbon = CType(RibbonApplicationMenu.Ribbon, C1TextEditorRibbon)
        ribbon.OpenDocument()
    End Sub
End Class

File: NetFramework/XHtmlEditor/CS/RichTextEditor/AppMenuTabs/OpenTab.cs
using RichTextEditor.AppMenuTabs.Items;
using System;
using System.Drawing;

namespace RichTextEditor.AppMenuTabs
{
    public partial class OpenTab : AppMenuTabFileList
    {
        public OpenTab()
        {
            Button.ItemImage = Properties.Resources.Open_large;
            Button.ItemText = Properties.Resources.Browse_txt;
            Caption = GetElementText(Properties.Resources.OpenTab_txt);
        }

        #region override
        protected override void OnMenuItemClick(AppMenuTabItem mi)
        {
            var ribbon = (C1TextEditorRibbon)RibbonApplicationMenu.Ribbon;
            ribbon.OpenDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text));
            mi.Item.Date = DateTime.Now;
        }

        protected override void OnMenuButtonClick()
        {
            var ribbon = (C1TextEditorRibbon)RibbonApplicationMenu.Ribbon;
            ribbon.OpenDocument();
        }
        #endregion        
    }
}
